### PR TITLE
status: Fix incorrect missing registry warning

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/openshift/origin/issues/5007

@deads2k

### CODE REVIEW

File: pkg/build/graph/analysis/bc.go [hunk 1]
```diff
@@ -28,6 +32,12 @@ bc:
 		for _, istNode := range g.SuccessorNodesByEdgeKind(bcNode, buildedges.BuildOutputEdgeKind) {
 			for _, uncastImageStreamNode := range g.SuccessorNodesByEdgeKind(istNode, imageedges.ReferencedImageStreamGraphEdgeKind) {
 				imageStreamNode := uncastImageStreamNode.(*imagegraph.ImageStreamNode)
+				key := MissingRequiredRegistryWarning
+				reason := missingRegistryMessage
+				if !imageStreamNode.IsFound {
```
Comment (deads2k) [inline pkg/build/graph/analysis/bc.go]: Why not just add the marker and continue inside of this `if`?  If the node isn't found, it won't have a `Status.DockerImageRepository` field.

Comment (kargakis) [inline pkg/build/graph/analysis/bc.go]: Missing image stream is another reason for a failed build push

Comment (deads2k) [inline pkg/build/graph/analysis/bc.go]: > Missing image stream is another reason for a failed build push

I'm just looking at code flow and looks to me like I'll hit this block and then later I'm guaranteed to enter the `if` block lower down and I'm always going to return the same thing.  I don't see why it couldn't simply be

```go
if !imageStreamNode.IsFound {
    markers = append(markers, osgraph.Marker{
        RelatedNodes: []graph.Node{istNode},
        Severity: osgraph.WarningSeverity,
        Key: MissingImageStreamWarning,
        Message: fmt.Sprintf(..., MissingImageStreamMessage)
    })

    continue
}

if len(imageStreamNode.Status.DockerImageRepository) == 0 {
    // existing code here
}
```

As it is now, I have to follow the flow and have some semantic understanding to realize that one if block is forcing behavior down below.

Comment (kargakis) [inline pkg/build/graph/analysis/bc.go]: Ok, updated plus added test for this

**Comment by kargakis** [comment]
[test]

**Comment by kargakis** [comment]
jenkins is green here too

**Comment by deads2k** [comment]
lgtm [merge]

**Comment by deads2k** [comment]
flake https://github.com/openshift/origin/issues/4242

re[merge]

**Comment by kargakis** [comment]
```
E1012 16:34:08.710289   24009 errors.go:56] apiserver received an error that is not an api.Status: test Error
E1012 16:34:09.369780   24009 apiserver.go:402] Failed to parse "not a timeout": time: invalid duration not a timeout
E1012 16:34:09.462148   24009 apiserver.go:361] apiserver was unable to write a JSON response: type apiserver.UnregisteredAPIObject is not registered for "version" and it will be impossible to Decode it, therefore Encode will refuse to encode it.
E1012 16:34:09.462343   24009 errors.go:56] apiserver received an error that is not an api.Status: type apiserver.UnregisteredAPIObject is not registered for "version" and it will be impossible to Decode it, therefore Encode will refuse to encode it.
--- FAIL: TestMaxInFlight (0.03s)
	handlers_test.go:50: unexpected response: 429
2015-10-12 16:34:09.884544 I | RoundTripper returned a response & error; ignoring response
FAIL
coverage: 66.9% of statements
FAIL	github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/apiserver	1.690s
```

re[merge]

**Comment by kargakis** [comment]
```
!!! Error in /data/src/github.com/openshift/origin/hack/update-generated-swagger-spec.sh:301
	'return 1' exited with status 1
Call stack:
	1: /data/src/github.com/openshift/origin/hack/update-generated-swagger-spec.sh:301 main(...)
Exiting with status 1
[FAIL] !!!!! Generate Failed !!!!
```
Why would I need to update swagger for this?

**Comment by kargakis** [comment]
And swagger doesn't produce any diff...re[merge]

**Comment by kargakis** [comment]
now integration tests :snowflake: :snowflake: :snowflake: 

...

**Comment by kargakis** [comment]
[merge]

**Pull request merged**